products/BellHybrid/services/time/AlarmOperations.cpp

```diff
@@ -347,6 +347,23 @@ namespace alarms
         AlarmOperationsCommon::handleAlarmEvent(event, alarmType, newStateOn);
     }
 
+    void AlarmOperations::turnOffPreWakeUp(IAlarmOperations::OnTurnOffPreWakeUp callback)
+    {
+        auto nextEvent = getNextPreWakeUpEvent();
+        if (nextEvent.isValid()) {
+            if (auto event = std::dynamic_pointer_cast<AlarmEventRecord>(nextEvent.parent); event) {
+                disablePreWakeUp(event);
```

Comment:
Doesn't work in the case where pre-wake up chime and pre-wake up frontlight have different time settings. From what I understand this function only affects active events (pre-wake up light or chime in progress).